artview/components/plot_radar.py

```diff
@@ -438,16 +438,23 @@ def _add_toolsBoxUI(self):
         toolmenu = QtWidgets.QMenu(self)
         toolZoomPan = toolmenu.addAction("Zoom/Pan")
         toolValueClick = toolmenu.addAction("Click for Value")
-        toolSelectRegion = toolmenu.addAction("Select a Region of Interest")
         toolReset = toolmenu.addAction("Reset Tools")
         toolDefault = toolmenu.addAction("Reset File Defaults")
         toolZoomPan.triggered.connect(self.toolZoomPanCmd)
         toolValueClick.triggered.connect(self.toolValueClickCmd)
-        toolSelectRegion.triggered.connect(self.toolSelectRegionCmd)
         toolReset.triggered.connect(self.toolResetCmd)
         toolDefault.triggered.connect(self.toolDefaultCmd)
+        self.toolmenu = toolmenu
         self.toolsButton.setMenu(toolmenu)
 
+    def add_mode(self, mode, label):
+        """ Add a tool entry with given label. Selecting that tool, execute
+        mode passing this component shared variables."""
+        def call_mode():
+            mode(self.get_sharedVariables())
+        action = self.toolmenu.addAction(label)
+        action.triggered.connect(call_mode)
+
```

Comment:
I like the embedded functions. I was thinking about this the other day and it might be useful for this package.